### ISSUE TITLE
VisualStyleRenderer.IsSupported returns false when a dialog is shown on Windows 10 v1809

### ISSUE THREAD

**Issue by creativbox** [dotnet/winforms#317]
* Have you experienced this same bug with .NET Framework?
Yes

**Problem description:**
The parent window does not render with styles on Windows 10 v1809 when an open or save file dialog is shown.

The native UXTheme function GetThemeAppProperties returns additional flags on Windows 10 v1809 if an dialog is opened (e.g. OpenFileDialog). The same flags are returned by the managed wrapper property Application.VisualStyleState. 
This causes the VisualStyleRenderer.IsSupported property to return false instead of true, because it does not check the return value of Application.VisualStyleState correctly:
```
bool supported =  (VisualStyleInformation.IsEnabledByUser &&
    (Application.VisualStyleState == VisualStyleState.ClientAreaEnabled ||
    Application.VisualStyleState == VisualStyleState.ClientAndNonClientAreasEnabled));
```
This check should consider additional flags (the VisualStyleState enum should be decorated with FlagsAttribute too).

**Comment by merriemcgaw** [comment]
We've seen this issue on .NET Framework as well and have fixed the issue there. @dreddy-work, have we brought the change into Core yet?

**Comment by dreddy-work** [comment]
We have fixed it in pr #322
